src/rwlock.rs

```diff
@@ -125,7 +128,10 @@ unsafe impl lock_api::RawRwLock for RawRwLock {
 
 //  --- X86 Specializations
 
-#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+#[cfg(all(
+    not(usync_tsan_enabled),
```

Comment:
Could this be removed, then change the x86 asm to use the tsan-friendly fence instead?

For example
```rs
let acquired = old_locked_bit == 0;
if acquired {
    fence_acquire(&self.state);
}

acquired
```
This clears up the TSAN errors on x86 for my side.

Reply:
Oh, yeah, that should work.